packages/history-service/src/internal/consumer-history.ts

```diff
@@ -17,6 +17,19 @@ export abstract class ConsumerHistory implements history.History {
       undefined,
       history.createLocation('/')
     );
+
+    /**
+     * The methods of `history.History` must be bound explicitly, because
+     * components like `Link` from the `react-router-dom` package deconstruct
+     * methods like `push` and `replace` from the `history`.
+     */
+    this.push = this.push.bind(this);
+    this.replace = this.replace.bind(this);
+    this.go = this.go.bind(this);
+    this.goBack = this.goBack.bind(this);
+    this.goForward = this.goForward.bind(this);
```

Comment:
BTW, could you explain, why do you define `goBack` and `goForward`, while [history](https://github.com/remix-run/history) defines [back](https://github.com/remix-run/history/blob/ba1cd3cf50465af8e1a89cc4f088c6372c10c12b/docs/api-reference.md#historyback) and [forward](https://github.com/remix-run/history/blob/ba1cd3cf50465af8e1a89cc4f088c6372c10c12b/docs/api-reference.md#historyforward)?